### PR TITLE
Add automation for the pktgen, hw provisioning, vswitch, and kubespray integration

### PR DESCRIPTION
**Pull request by denverwilliams**
## Description
1. Adds automation for the pktgen, hw provisioning, vswitch, and kubespray integration.
2. Adds a Makefile with a pre-defined configuration that will reproduce the testbed in a CI environment.

Issues:
vulk/cncf_cnf-testbed#36

## How Has This Been Tested?
* [x]  Tested locally
* [ ]  Have not tested

## Types of changes
* [ ]  Bug fix (non-breaking change which fixes an issue)
* [x]  New feature (non-breaking change which adds functionality)
* [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
 
## Checklist:
* [x]  My change requires a change to the documentation.
* [x]  I have updated the documentation accordingly.
* [ ] No updates required.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
-# WORKER_COUNT=${WORKER_COUNT:-1}
-# PACKET_FACILITY=${PACKET_FACILITY:-sjc1}
-# VLAN_SEGMENT=${VLAN_SEGMENT:-$DEPLOY_NAME}
-# PLAYBOOK=${PLAYBOOK:-k8s_worker_vswitch_quad_intel.yml}
-# RELEASE_TYPE=${RELEASE_TYPE:-stable}
-# HOSTS_FILE=${HOSTS_FILE:-$(pwd)/data/$DEPLOY_NAME/nodes.env}
+#Vars req for running
+export PACKET_AUTH_TOKEN=YOUR_PACKET_AUTH_TOKEN
+export PROJECT_ID=YOUR_PACKET_PROJECT_ID
+export PACKET_PROJECT_NAME=YOUR_PACKET_PROJECT_NAME
```
Comment (michaelspedersen) [inline Makefile]: These exports probably needs to be removed. I was unable to run with the env vars set outside of the Makefile, when they were also present in here.

File: tools/packet_generator_provisioning.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+NIC_TYPE=${NIC_TYPE:--e quad_intel=true}
+FACILITY=${FACILITY:-sjc1}
+VLAN_SEGMENT=${VLAN_SEGMENT:-$DEPLOY_NAME}
+PLAYBOOK=${PLAYBOOK:-packet_generator.yml}
+
+if ! [ -z ${PKTGEN_HOSTS+x} ]; then
+    HOSTS="$PKTGEN_HOSTS,"
+    PKTGEN_IPS_ARRAY=($(echo $PKTGEN_HOSTS | tr ',' ' '))
+elif ! [ -z ${NODE_FILE+x} ]; then
+    PKTGEN_IPS_ARRAY=($(yq r $NODE_FILE nodes.[*].addr | tr -d '\n''-'))
```
Comment (michaelspedersen) [inline tools/packet_generator_provisioning.sh]: `yq` must be installed on the machine running the deployment

File: tools/hardware_provisioning.sh [hunk 3]
```diff
@@ -1,23 +1,35 @@
...
+
+NODE_GROUP_ONE_NAME=${NODE_GROUP_ONE_NAME:-$DEPLOY_NAME-master}
+NODE_GROUP_TWO_NAME=${NODE_GROUP_TWO_NAME:-$DEPLOY_NAME-worker}
+NODE_GROUP_ONE_COUNT=${NODE_GROUP_ONE_COUNT:-1}
+NODE_GROUP_TWO_COUNT=${NODE_GROUP_TWO_COUNT:-1}
+FACILITY=${FACILITY:-sjc1}
+NODE_GROUP_ONE_DEVICE_PLAN=${NODE_GROUP_ONE_DEVICE_PLAN:-m2.xlarge.x86}
+NODE_GROUP_TWO_DEVICE_PLAN=${NODE_GROUP_TWO_DEVICE_PLAN:-n2.xlarge.x86}
+OPERATING_SYSTEM=${OPERATING_SYSTEM:-ubuntu_16_04}
+
```
Comment (michaelspedersen) [inline tools/hardware_provisioning.sh]: hardware-provisioning.env.example needs to be updated to match these new variable names

File: tools/packet_generator_provisioning.sh (hunk 2 above)
Comment (denverwilliams) [reply tools/packet_generator_provisioning.sh]: This commit should resolve this: 
https://github.com/cncf/cnf-testbed/pull/320/commits/71be8552ca212f7d60ea4e6293c21a49b9c6d5cf

File: Makefile (hunk 1 above)
Comment (denverwilliams) [reply Makefile]: Resolved by:
https://github.com/cncf/cnf-testbed/commit/ccf22f5ad60194dc14d2273f96525a5f74fdff5d

File: tools/hardware_provisioning.sh (hunk 3 above)
Comment (denverwilliams) [reply tools/hardware_provisioning.sh]: Resolved by:
https://github.com/cncf/cnf-testbed/commit/f9b8c75a065a96853a4d4f2db54ee30687976f3e

**Comment by michaelspedersen** [comment]
+1 Looks good. Haven't done any tests based on most recent updates, but from last review it should work "out of the box" with the proposed changes in place

File: tools/README.md [hunk 4]
```diff
@@ -18,12 +18,17 @@ found in hardware_provisioning.sh
...
-WORKER_PLAN=${WORKER_PLAN:-n2.xlarge.x86}
+STATE_FILE=${STATE_FILE:-$(pwd)/data/$DEPLOY_NAME/terraform.tfstate}
+NODE_FILE=${NODE_FILE:-$(pwd)/data/$DEPLOY_NAME/kubernetes.env}
+
+NODE_GROUP_ONE_NAME=${NODE_GROUP_ONE_NAME:-$DEPLOY_NAME-master}
+NODE_GROUP_TWO_NAME=${NODE_GROUP_TWO_NAME:-$DEPLOY_NAME-worker}
+NODE_GROUP_ONE_COUNT=${NODE_GROUP_ONE_COUNT:-1}
+NODE_GROUP_TWO_COUNT=${NODE_GROUP_TWO_COUNT:-1}
+FACILITY=${PACKET_FACILITY:-sjc1}
+NODE_GROUP_ONE_DEVICE_PLAN=${NODE_GROUP_ONE_DEVICE_PLAN:-m2.xlarge.x86}
```
Comment (wavell) [inline tools/README.md]: Is there any way to get away from hard coding the machine types?

Comment (denverwilliams) [reply tools/README.md]: These are default variables, so can be overridden.

**Pull request merged**